src/main/ngapp/accio/src/app/editor/magic-wand.service.ts

```diff
@@ -298,12 +212,54 @@ export class MagicWandService {
         }
         // Visits the pixel and check if it should be part of the mask.
         visited.add(this.coordToDataArrayIndex(x, y, imgData.width));
-        if (this.getIsScribbleMask(scribbles, imgData, neighborPixel, tolerance)) {
+        // if (this.getIsScribbleMask(scribbles, imgData, neighborPixel, tolerance)) {
+        //   visit.push(neighborPixel);
+        // }
+        if (tree.nearest(
+            this.scribblesToColors(new Set([
+            this.coordToDataArrayIndex(x, y, imgData.width)]), imgData)[0],
+            1, tolerance * tolerance).length > 0) {
           visit.push(neighborPixel);
         }
       }
     }  // End of while loop.
     
     return mask;
   }
+
+  /**@returns {number} the straight line distance between the two colors
+   * @param {Array<number> [R, G, B]} basisColor and 
+   * @param {Array<number> [R, G, B]} secondColor
+   */
+  rgbEuclideanDist(colorA: Color, colorB: Color): number {
+
+    let distance = 0;
+
+    for (let attribute in colorA) {
+      distance += Math.pow((colorA[attribute] - colorB[attribute]), 2);
+    }
```

Comment:
Since this is going to be called a lot, we might want to optimize a little. Also, I find the loop a little unclear -- and it could do surprising things if there were extra properties. I'd suggest calculating dr, dg, and db like you do below and returning dr*dr + dg*dg + db*db.